src/hooks/useKanban.js

```diff
@@ -105,12 +120,23 @@ export function useKanban() {
     try {
       const batch = writeBatch(db);
       
-      Object.values(newColumns).forEach((column) => {
+      // Converte o objeto em array e ordena pelo índice
+      const columnsArray = Object.entries(newColumns).map(([id, column]) => ({
+        id,
+        ...column
+      }));
+
+      // Atualiza a ordem de cada coluna
+      columnsArray.forEach((column, index) => {
         const columnRef = doc(db, 'boards/main/columns', column.id);
-        batch.update(columnRef, { order: column.order });
+        batch.update(columnRef, { 
+          order: index,
+          updatedAt: new Date()
+        });
       });
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
Object.entries(newColumns).forEach(([id, column], index) => {
  const columnRef = doc(db, 'boards/main/columns', id);
  batch.update(columnRef, { 
    order: index,
    updatedAt: new Date()
  });
});
```
The columnsArray mapping operation could be optimized by combining the map and subsequent forEach into a single iteration to reduce complexity from O(2n) to O(n).



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;